v8py/jsobject.cpp

```diff
@@ -113,12 +124,32 @@ int js_object_setattro(js_object *self, PyObject *name, PyObject *value) {
 
     IN_V8;
     IN_CONTEXT(self->context.Get(isolate));
+    JS_TRY
 
     Local<Object> object = self->object.Get(isolate);
-    if (!object->Set(context, js_from_py(name, context), js_from_py(value, context)).FromJust()) {
-        PyErr_SetString(PyExc_AttributeError, "Object->Set completely failed for some reason");
+
+    double timeout = 0;
+    {
+        context_c *ctx_c = (context_c *) context->GetEmbedderData(CONTEXT_OBJECT_SLOT).As<External>()->Value();
+        timeout = ctx_c->timeout;
+    }
+
+    if (!setup_timeout(timeout)) return -1;
+
+    Maybe<bool> result = value ?
+        object->Set(context, js_from_py(name, context), js_from_py(value, context)) :
+        object->Delete(context, js_from_py(name, context));
+
+    if (!cleanup_timeout(timeout)) return -1;
+
+    PY_PROPAGATE_JS_RET(-1);
+
+    if (result.IsNothing() || !result.FromJust()) {
```

Comment:
There's no need to check `IsNothing()` here, if an exception was thrown it would have been handled by the PY_PROPAGATE_JS_RET.

Reply:
Also, I'm really not sure why this would return Just(false). If the attribute doesn't exist, it would throw an exception.

Reply:
In fact, a quick look at the v8 code proves that both these methods can never return Just(false), they only ever return Just(true) or Nothing. And they only return Nothing when the set or delete throws an exception or execution was terminated, both of which are handled by PY_PROPAGATE_JS_RET, so this entire if statement never triggers. https://github.com/v8/v8/blob/4a26804cf06833108d048800233a4899f2d04f96/src/api.cc#L4337 https://github.com/v8/v8/blob/4a26804cf06833108d048800233a4899f2d04f96/src/api.cc#L4850